jest/index.d.ts

```diff
@@ -147,11 +190,28 @@ declare namespace jest {
         new (...args: any[]): any
     }
 
-    interface Mock<T> extends Function {
+    interface Mock<T> extends Function, MockInstance<T> {
         new (): T;
         (...args: any[]): any;
+    }
+
+    /**
+     * Wrap module with mock definitions
+     * @example
+     *  jest.mock("../api");
+     *  import { Api } from "../api";
+     *
+     *  const myApi: jest.Mocked<Api> = new Api() as any;
+     *  myApi.myApiMethod.mockImplementation(() => "test");
+     */
+    type Mocked<T> = {
+        [P in keyof T]: T[P] & MockInstance<T>;
```

Comment:
Should this have been `MockInstance<T[P]>` instead?  Right now it's picking up the type of the parent object instead of the individual methods.